lib/internal/encoding.js

```diff
@@ -426,11 +420,7 @@ function makeTextDecoderICU() {
       if (options !== null)
         flags |= options.stream ? 0 : CONVERTER_FLAGS_FLUSH;
 
-      const ret = _decode(this[kHandle], input, flags);
-      if (typeof ret === 'number') {
-        throw new ERR_ENCODING_INVALID_ENCODED_DATA(this.encoding, ret);
```

Comment:
Can you rebase from the main and I remove the unnecessary commit?

Reply:
Done.